internal/cli/pkg/cluster/create.go

```diff
@@ -0,0 +1,331 @@
+	backupVolume := buildBackupVolume(param.BackupVolume)
+	parameters := buildOBClusterParameters(param.Parameters)
+	topology := buildOBClusterTopology(param.Topology)
+	obcluster := &v1alpha1.OBCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:   param.Namespace,
+			Name:        param.Name,
+			Annotations: map[string]string{},
+		},
+		Spec: v1alpha1.OBClusterSpec{
+			ClusterName:      param.ClusterName,
+			ClusterId:        param.ClusterId,
+			OBServerTemplate: observerTemplate,
+			MonitorTemplate:  monitorTemplate,
+			BackupVolume:     backupVolume,
+			Parameters:       parameters,
+			Topology:         topology,
+			UserSecrets:      generateUserSecrets(param.Name, param.ClusterId),
+		},
+	}
+	switch param.Mode {
+	case string(modelcommon.ClusterModeStandalone):
+		obcluster.Annotations[oceanbaseconst.AnnotationsMode] = oceanbaseconst.ModeStandalone
+	case string(modelcommon.ClusterModeService):
+		obcluster.Annotations[oceanbaseconst.AnnotationsMode] = oceanbaseconst.ModeService
+	default:
+	}
+	return obcluster
+}
+func NewCreateOptions() *CreateOptions {
+	return &CreateOptions{
+		OBServer: &param.OBServerSpec{
+			Storage: &param.OBServerStorageSpec{},
+		},
+		Parameters: make([]modelcommon.KVPair, 0),
+		Zones:      make(map[string]string),
+		Topology:   make([]param.ZoneTopology, 0),
+	}
+}
+
+type CreateOptions struct {
+	BaseOptions
+	ClusterName  string               `json:"clusterName"`
+	ClusterId    int64                `json:"clusterId"`
+	RootPassword string               `json:"rootPassword"`
+	Topology     []param.ZoneTopology `json:"topology"`
+	OBServer     *param.OBServerSpec  `json:"observer"`
+	Monitor      *param.MonitorSpec   `json:"monitor"`
+	Parameters   []modelcommon.KVPair `json:"parameters"`
+	BackupVolume *param.NFSVolumeSpec `json:"backupVolume"`
+	Zones        map[string]string    `json:"zones"`
+	Mode         string               `json:"mode"`
+}
+
+func (o *CreateOptions) Validate() error {
+	if !CheckPassword(o.RootPassword) {
+		return fmt.Errorf("Password is not secure, must contain at least 2 uppercase and lowercase letters, numbers and special characters")
+	}
+	if !CheckResourceName(o.Name) {
+		return fmt.Errorf("invalid resource name in k8s: %s", o.Name)
+	}
+	return nil
+}
+
+// Parse Cli args and set options
+func (o *CreateOptions) Parse() error {
+	topology, err := mapZonesToTopology(o.Zones)
+	if err != nil {
+		return err
+	}
+	o.Topology = topology
+	// if not specific id, using timestamp
+	if o.ClusterId == 0 {
+		o.ClusterId = time.Now().Unix() % 4294901759
+	}
+	// if not specific password, using random password
+	if o.RootPassword == "" {
+		o.RootPassword = generateRandomPassword()
+	}
+	// if not specific name, using cluster name
+	if o.Name == "" {
+		o.Name = o.ClusterName
+	}
+	return nil
+}
+
+// AddZoneFlags adds the zone-related flags to the command.
+func (o *CreateOptions) AddZoneFlags(cmd *cobra.Command) {
```

Comment:
maybe it's more reasonable to move functions deal with flags into another package